awc/src/request.rs

```diff
@@ -23,8 +23,17 @@ use crate::ClientConfig;
 
 #[cfg(any(feature = "flate2-zlib", feature = "flate2-rust"))]
 const HTTPS_ENCODING: &str = "br, gzip, deflate";
-#[cfg(not(any(feature = "flate2-zlib", feature = "flate2-rust")))]
+#[cfg(all(
+    not(any(feature = "flate2-zlib", feature = "flate2-rust")),
+    feature = "compress"
+))]
 const HTTPS_ENCODING: &str = "br";
+#[cfg(not(any(
+    feature = "flate2-zlib",
+    feature = "flate2-rust",
+    feature = "compress"
+)))]
+const HTTPS_ENCODING: &str = "identity";
```

Comment:
would it be clearer to use `cfg-if` ?

Reply:
Do you mean https://docs.rs/cfg-if/1.0.0/cfg_if ? I can't find anythin like `cfg-it` it under https://doc.rust-lang.org/reference/conditional-compilation.html

Reply:
That looks nice indeed and its already in the dependency tree. I'll change it.

Reply:
should avoid any situations where a combination of feature flags causes compile errors

Reply:
Done. The solution i originally had i mind doesn't work with the macro unfortunately, but it allows to reduce the complexity of the cfg attributes, so its still a win i guess.
```rust
const HTTPS_ENCODING: &str = cfg_if::cfg_if! {
    if #[cfg(any(feature = "flate2-zlib", feature = "flate2-rust"))] {
         "br, gzip, deflate";
    } else if #[cfg(feature = "compress")] {
        "br";
    } else {
        "identity";
    }
};
```